### PR TITLE
Added new fields in hostgroup entity

### PR DESCRIPTION
**Pull request by sghai**
Added two new params for hostgroup entity. `puppet_proxy` and `puppet_ca_proxy`

### CODE REVIEW

**Comment by sghai** [comment]
As per the API docs, one more we should add i.e. `content-source`. When I tried to add this too, I got following:

`MissingValueError: Cannot find a value for the "content_source" field. Searched for keys named ('content_source', 'content_source_id'), but available keys are ['content_view_id', u'environment_id', u'all_puppetclasses', u'updated_at', u'realm_name', u'operatingsystem_name', u'domain_id', u'ptable_id', u'parameters', u'title', u'subnet_id', u'environment_name', u'architecture_name', u'domain_name', u'id', 'parent_id', u'subnet_name', u'architecture_id', u'medium_name', u'operatingsystem_id', u'config_groups', u'compute_profile_name', u'locations', u'template_combinations', u'puppet_ca_proxy_id', u'puppetclasses', u'puppet_proxy_id', u'ptable_name', u'organizations', u'name', u'medium_id', u'created_at', 'lifecycle_environment_id', u'realm_id', u'compute_profile_id'].`

**Comment by sghai** [comment]
Test results for added fields are:
```
In [5]: host_group = entities.HostGroup(server_config, name="hg22", puppet_proxy="1",  puppet_ca_proxy="1").create()

In [8]: host_group.name
Out[8]: u'hg22'

In [9]: host_group.puppet_proxy
Out[9]: nailgun.entities.SmartProxy(nailgun.config.ServerConfig(url='https://cloud-qe-1.idmqe.lab.eng.bos.redhat.com', verify=False, auth=('admin', 'changeme')), id=1)

In [10]: host_group.puppet_ca_proxy
Out[10]: nailgun.entities.SmartProxy(nailgun.config.ServerConfig(url='https://cloud-qe-1.idmqe.lab.eng.bos.redhat.com', verify=False, auth=('admin', 'changeme')), id=1)
```

**Comment by oshtaier** [comment]
So `content_source` works for you or not? Because you showed example of error, but anyway submitting it in PR

**Comment by sghai** [comment]
I'm getting value Error. However, as per API doc `content_source` is a valid name. I submitted it in PR because I need that for upcoming robottelo tests and that's optional not a required field.

**Comment by oshtaier** [comment]
In my opinion, it will be nice to figure out reason of such error before submitting this PR, because you will not be able to use that field in Robottelo anyway.

**Comment by sghai** [comment]
Yeah. :)
I'm not sure the reason why this error is appearing. Will ask @Ichimonji10 if he has some suggestion. Thanks.

**Comment by Ichimonji10** [comment]
This looks nice. If we can take care of the exception, I'm happy to merge.

For the record:

> \<ichimonji10> sghai: What happens is, after NailGun submits a POST request (to create an entity on the server), NailGun expects the server to return a full description of the just-created entity.
> \<ichimonji10> sghai: So NailGun is expecting Satellite to say "I just created a new hostgroup! Here's what the new hostgroup looks like."
> \<ichimonji10> sghai: But Satellite isn't returning a full description. Satellite doesn't return any value for the "content_source" field.

**Comment by sghai** [comment]
```
In [5]: host_group = entities.HostGroup(server_config, name="hg114", puppet_proxy="1",  puppet_ca_proxy="1", content_source="1").create()

In [6]: host_group.content_source
Out[6]: nailgun.entities.SmartProxy(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), id=1)
```

File: nailgun/entities.py [hunk 1]
```diff
@@ -1799,6 +1802,16 @@ def __init__(self, server_config=None, **kwargs):
         self._meta = {'api_path': 'api/v2/hostgroups', 'server_modes': ('sat')}
         super(HostGroup, self).__init__(server_config, **kwargs)
 
+    def create(self, create_missing=None):
+        """Do extra work to fetch a complete set of attributes for this entity.
+        For more information, see `Bugzilla #1223540
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1223540>`_.
```
Comment (Ichimonji10) [inline nailgun/entities.py]: This bug actually references compute resources, not host groups. Can you drop this bit from the docstring and file a github issue saying "file a BZ for `HostGroup.create` and link to it"?

File: nailgun/entities.py [hunk 2]
```diff
@@ -1830,7 +1843,9 @@ def read(self, entity=None, attrs=None, ignore=None):
                 self._server_config,
                 id=attrs['id']
             ).update_json([])
-            for attr in ('content_view_id', 'lifecycle_environment_id'):
+            for attr in ('content_view_id',
+                         'lifecycle_environment_id',
+                         'content_source_id'):
```
Comment (Ichimonji10) [inline nailgun/entities.py]: Aha. So this attribute is not returned in response to either `POST` or `GET` requests. :cry:

**Comment by Ichimonji10** [comment]
Satellite returns `content_source_id` in response to PUT requests. We can nicely work around the issue by using this fact. Here, let me try to put together something...

**Comment by Ichimonji10** [comment]
ACK pending [this one comment](https://github.com/SatelliteQE/nailgun/pull/194#discussion_r38535542).

**Comment by rplevka** [comment]
nitpick: could you sort the dict. keys in an alphabetical order? (absolutely not a blocker)

File: nailgun/entities.py (hunk 1 above)
Comment (sghai) [inline nailgun/entities.py]: Yeah..Filed issue: https://github.com/SatelliteQE/nailgun/issues/197

**Comment by sghai** [comment]
Guys: Thanks for review comments. I think now PR is ready for merge. Please take another look. Thanks

**Comment by oshtaier** [comment]
ACK

**Comment by elyezer** [comment]
:+1:

**Pull request merged**